pandas/_libs/lib.pyx

```diff
@@ -107,24 +109,57 @@ def memory_usage_of_objects(arr: object[:]) -> int64_t:
 def is_scalar(val: object) -> bint:
     """
     Return True if given value is scalar.
-
+    
+    Parameters
+    ----------
+    val : object
     This includes:
-    - numpy array scalar (e.g. np.int64)
-    - Python builtin numerics
-    - Python builtin byte arrays and strings
-    - None
-    - instances of datetime.datetime
-    - instances of datetime.timedelta
-    - Period
-    - instances of decimal.Decimal
-    - Interval
-    - DateOffset
+    * numpy array scalar (e.g. np.int64)
+    * Python builtin numerics
+    * Python builtin byte arrays and strings
+    * None
+    * instances of datetime.datetime
+    * instances of datetime.timedelta
+    * Period
+    * instances of decimal.Decimal
+    * Interval
+    * DateOffset
+    * Fraction
+    * Number
 
+    Returns
+    -------
+     a bool object.
+    
+    Examples
+    --------
+    >>> dt = pd.datetime.datetime(2018, 10, 3)
+    >>> pd.is_scalar(dt)
+    True
+
+    >>> pd.api.types.is_scalar([2, 3])
+    False
+    
+    >>> pd.api.types.is_scalar({0:1, 2:3})
+    False
+    
+    >>> pd.api.types.is_scalar((0, 2))
+    False
+    
+    pandas supports PEP 3141 numbers:
+    
+    >>> from fractions import Fraction
+    >>> pd.api.types.is_scalar(Fraction(3, 5))
+    True
+
+    >>> from numbers import Number
+    >>> pd.api.types.is_scalar(Number())
```

Comment:
I think it makes more sense to have an example `Number` here. Not sure what an empty `Number` instance can be used for, but doesn't seem something you'd use as often as an actual number.

Reply:
What should I put in here, then, @datapythonista ?
Any number would fall into `int` category... :/

Reply:
yeah prob can remove this last Number example, its not generally useful unless you actually inherit from Number of a subclass.